### ISSUE TITLE
p2p communication should strongly typed

### ISSUE THREAD

**Issue by wxf4150** [omnilaboratory/obd#49]
The data type of p2p communication is not strongly typed. At present, using "~" to split the data stream, using gjson to extract json data is not safe enough; it should be possible to use protobuf to do complete and strongly typed data extraction. The data stream separator should be longer to reduce potential segmentation errors,  maybe three "~" char will better.

**Comment by neocarmack** [comment]
yes. It is a problem. We are now integrating with the latest omnicore, this issue should be fixed accordingly.
